### DISCUSSION TITLE
What is the monitor of lr_scheduler?

### DISCUSSION THREAD

**Discussion by Co1lin** [Lightning-AI/pytorch-lightning#8554] [General]
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/8541

<div type='discussions-op-text'>

<sup>Originally posted by **Co1lin** July 24, 2021</sup>
Hi! I’m wondering what is `metric_to_track` for the parameter monitor  of a learning rate scheduler? I see the example that uses `"val_loss"` as its value, but I don’t know is `val_loss` a name of a variable (where should I declare this variable?) or a keyword that can be recognized by pytorch_lightning such that it will monitor the loss of the `validation_step`?</div>

I get a warning saying:

```shell
packages/pytorch_lightning/callbacks/model_checkpoint.py:610: LightningDeprecationWarning: Relying on `self.log('val_loss', ...)` to set the ModelCheckpoint monitor is deprecated in v1.2 and will be removed in v1.4. Please, create your own `mc = ModelCheckpoint(monitor='your_monitor')` and use it as `Trainer(callbacks=[mc])`.
```
I'm not sure whether this usage will be deprecated for scheduler's monitor too.

**Comment by ethanwharris** [comment]
Hi @Co1lin, you can configure the monitor for LR scheduling in the `configure_optimizers` hook of your `LightningModule` as detailed here: https://pytorch-lightning.readthedocs.io/en/stable/common/optimizers.html?highlight=lr%20scheduler#learning-rate-scheduling

`metric_to_track` is the name of the logged metric to use as the monitor as a string (similarly for "val_loss"). So when you then call `self.log("metric_to_track", my_metric_value)` that is the metric that will be used. The deprecation for ModelCheckpoint relates to the fact that we used to just take whichever value was logged as the `val_loss` as the quantity to monitor. So the usage for LR scheduling hasn't changed, we just give a warning for users who rely on the behaviour for checkpointing.

Hope that helps :smiley:
